test/testcafe/spec/ChallengeAuthenticatorPhone_spec.js

```diff
@@ -0,0 +1,240 @@
+    await challengeFactorPageObject.clickElement('.phone-authenticator-verify__button--secondary');
+    const pageSubtitle = challengeFactorPageObject.getFormSubtitle();
+    await t.expect(pageSubtitle).contains('A code was sent to');
+    await t.expect(pageSubtitle).contains('Enter the code below to verify.');
+    await t.expect(challengeFactorPageObject.getSaveButtonLabel('input[type="submit"]')).eql('Verify');
+    await t.expect(challengeFactorPageObject.elementHasClass('.o-form-button-bar', 'hide')).eql(false);
+    await t.expect(challengeFactorPageObject.elementHasClass('.o-form-fieldset', 'hide')).eql(false);
+    await t.expect(challengeFactorPageObject.elementHasClass('.phone-authenticator-verify__button--primary', 'hide')).eql(true);
+    await t.expect(challengeFactorPageObject.elementHasClass('.phone-authenticator-verify__button--secondary', 'hide')).eql(true);
+  });
+
+test
+  .requestHooks(voiceOnlyMock)(`Voice Primary - clicking on primary button changes view`, async t => {
+    const challengeFactorPageObject = await setup(t);
+    await t.expect(challengeFactorPageObject.elementHasClass('.o-form-button-bar', 'hide')).eql(true);
+    await t.expect(challengeFactorPageObject.elementHasClass('.o-form-fieldset', 'hide')).eql(true);
+    await t.expect(challengeFactorPageObject.elementHasClass('.phone-authenticator-verify__button--primary', 'hide')).eql(false);
+    await t.expect(challengeFactorPageObject.elementExists('.phone-authenticator-verify__button--secondary')).eql(false);
+    await challengeFactorPageObject.clickElement('.phone-authenticator-verify__button--primary');
+    const pageSubtitle = challengeFactorPageObject.getFormSubtitle();
+    await t.expect(challengeFactorPageObject.getSaveButtonLabel('input[type="submit"]')).eql('Verify');
+    await t.expect(pageSubtitle).contains('Calling');
+    await t.expect(pageSubtitle).contains('Enter the code below to verify.');
+    await t.expect(challengeFactorPageObject.elementHasClass('.o-form-button-bar', 'hide')).eql(false);
+    await t.expect(challengeFactorPageObject.elementHasClass('.o-form-fieldset', 'hide')).eql(false);
+    await t.expect(challengeFactorPageObject.elementHasClass('.phone-authenticator-verify__button--primary', 'hide')).eql(true);
+    await t.expect(challengeFactorPageObject.elementExists('.phone-authenticator-verify__button--secondary')).eql(false);
+  });
+
+test
+  .requestHooks(invalidCodeMock)(`Entering invalid passcode results in an error`, async t => {
+    const challengeFactorPageObject = await setup(t);
+    await challengeFactorPageObject.clickElement('.phone-authenticator-verify__button--primary');
+    await challengeFactorPageObject.verifyFactor('credentials.passcode', 'abcd');
+    await challengeFactorPageObject.clickNextButton();
+    await challengeFactorPageObject.waitForErrorBox();
+    await t.expect(challengeFactorPageObject.getInvalidOTPError()).contains('Authentication failed');
```

Comment:
Would be better to verify request body like https://github.com/okta/okta-signin-widget/blob/master/test/testcafe/spec/ChallengeAuthenticatorSecurityQuestion_spec.js#L45
Same for other case we made remediation requests.

Reply:
Will revisit this once the API remediation is ready..